server/src/main/java/io/crate/planner/operators/CorrelatedJoin.java

```diff
@@ -162,12 +163,24 @@ public Void visitOuterColumn(OuterColumn outerColumn, Void ignored) {
         if (inputPlan == newInputPlan) {
             return this;
         }
+        // Original outputs can contain unused columns.
+        // We should update outputs after the prunning to avoid the edge case
+        // when a column is still referenced in CorrelatedJoin but not collected anymore after the prunning.
+        // See https://github.com/crate/crate/issues/1590 for details.
```

Comment:
This comment is more internal and doesn't fit to a change entry.

Still don't have a simple example that I could use for a test/change entry. 

I confirmed that query in the reported issue works as expected with this change.

@mkleen, @mfussenegger may I get an early feedback on this? If direction is okay, I will continue looking for a simpler repro for a test/change entry.

Reverting https://github.com/crate/crate/pull/13736/ or https://github.com/crate/crate/pull/13895 is not an options as we don't want to get a performance regression. In general, it works because CorrelatedJoin only references actually collected columns.

Some workarounds mentioned in the issue force `unused_field` from the example query be collected.

Reply:
I think column-pruning should also be applied to the outputs of the CorrelatedJoin operator. This can be tested in isolation but testing `pruneOutputsExcept`.

Reply:
Do you mean smth in addition to the current change (which is basically an implicit prunning) or replacing it with smth like

```
return new CorrelatedJoin(
            newInputPlan,
            selectSymbol,
            subQueryPlan,
            outputs
        ).pruneOutputsExcept(outputsToKeep); 
``` ?

Reply:
The change as it is applies to pruning to the outputs.

Reply:
So the current approach looks ok for me.

Reply:
The direction is okay but I think the comments can be removed. That's just re-explaining what a proper `pruneOutputsExcept` implementation needs to do, and is documented on the itnerface.